### DISCUSSION TITLE
Prisma is lowering column alias while using $queryRaw and $queryRawUnsafe

### DISCUSSION THREAD

**Discussion by RomanSkrypnik** [prisma/prisma#19718] [Q&A] [answered]
### Bug description

I want to make transform my column name from snake case to camel case using SQL query alias
Unfortunately, Prisma is lowering each letter in each property of object

### How to reproduce

1. Create model: 

```
model User {
  id        Int      @id @unique @default(autoincrement())
  email     String   @unique @db.VarChar(50)
  password  String   @db.VarChar(255)
  createdAt DateTime @default(now()) @map("created_at")
  signinAt  DateTime @default(now()) @map("signin_at")
  status    Status   @default(ACTIVE)

  @@map("users")
}
```

2. Use $queryRaw or $queryRawUnsafe with camel case alias:

<img width="242" alt="image" src="https://github.com/prisma/prisma/assets/44501848/607f1b4c-259c-4c87-a677-83096e81a468">

3. Watch the result:
<img width="307" alt="image" src="https://github.com/prisma/prisma/assets/44501848/5e94c1eb-9f35-481d-ba28-2ff2767a859d">

### Expected behavior

Prisma should not affect aliases while using $queryRaw or $queryRawUnsafe

### Prisma information

```
model User {
  id        Int      @id @unique @default(autoincrement())
  email     String   @unique @db.VarChar(50)
  password  String   @db.VarChar(255)
  createdAt DateTime @default(now()) @map("created_at")
  signinAt  DateTime @default(now()) @map("signin_at")
  status    Status   @default(ACTIVE)

  @@map("users")
}
```

### Environment & setup

OS: MacOS 12.6
Database: PostgreSQL
Node.js version: 18.12.1

### Environment variables loaded from .env

```
prisma                  : 4.15.0
@prisma/client          : 4.15.0
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 8fbc245156db7124f997f4cecdd8d1219e360944 (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 8fbc245156db7124f997f4cecdd8d1219e360944 (at node_modules/@prisma/engines/migration-engine-darwin)
Format Wasm             : @prisma/prisma-fmt-wasm 4.15.0-28.8fbc245156db7124f997f4cecdd8d1219e360944
Default Engines Hash    : 8fbc245156db7124f997f4cecdd8d1219e360944
Studio                  : 0.484.0

```

**Comment by janpio** [comment, marked as answer]
Can you share the SQL of the raw query you are running?
Are you quoting the column names properly? Otherwise, PostgreSQL might lowercase those for you.

**Reply by RomanSkrypnik** [reply]
> Can you share the SQL of the raw query you are running? Aer you quoting the column names properly? Otherwise PostgreSQL might lowercase those for you.

<img width="566" alt="image" src="https://github.com/prisma/prisma/assets/44501848/5aa82b74-bd7e-4cd0-a469-b03d417fa401">
<img width="125" alt="image" src="https://github.com/prisma/prisma/assets/44501848/e82aad8d-69c9-4fae-b0f0-d983658a45ab">

**Reply by janpio** [reply]
Add `"` around the column aliases, like `u.created_at as "createdAt"`. This will return `createdAt` instead of `createdat` for the query.

**Reply by nurul3101** [reply]
I am assuming this issue is resolved now.

**Comment by karis** [comment]
hi every one..

im asking to reopen this case,
or may be it will be created new issue.

here in my case:
in my orders table i am not using @map feature
below is my table prisma.schema

```
model Orders {
  id          String      @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  payment       Int?        @default(0)
  changes     Int?        @default(0)
  discount    Int?        @default(0)
  total       Int?        @default(0)
  createdAt   DateTime?   @default(now()) @db.Timestamptz(6)
  orderItem   OrderItem[]
  storeId    String?     @db.Uuid
}
```

then i want to select  group by from those table
```
const dataResult = await prisma.$queryRawUnsafe('SELECT ord.createdAt date, sum(ord.total) as grandTotal FROM public."Orders" as ord where ord.storeId = $1 group by createdAt', storeId)
```

then i got, these report storeId in lower case
```
Raw query failed. Code: `42703`. Message: `column ord.storeid does not exist`
  at si.handleRequestError (./node_modules/.pnpm/@prisma+client@5.7.1_prisma@5.7.1/node_modules/@prisma/client/runtime/library.js:125:6817)  
  at si.handleAndLogRequestError (./node_modules/.pnpm/@prisma+client@5.7.1_prisma@5.7.1/node_modules/@prisma/client/runtime/library.js:125:6151)  
  at si.request (./node_modules/.pnpm/@prisma+client@5.7.1_prisma@5.7.1/node_modules/@prisma/client/runtime/library.js:125:5859)  
```

thanks for your help

**Reply by janpio** [reply]
You have the same problem as explained above, you need to quote identifiers with upercase letters or they will be lowercased on executed. That is also why `public."Orders"` is with `"` or otherwise it would be executed as `public.order`.

**Reply by karis** [reply]
owh i miss the quote...

thank you, 
it works...
